concourse/scripts/bosh_ensure_uploaded.sh

```diff
@@ -0,0 +1,31 @@
+#! /bin/sh
+set -e
+
+script_dir=$(cd "$(dirname "$0")" && pwd)
+
+type="${TYPE:-$1}"
+name="${NAME:-$2}"
+#shellcheck disable=SC2153
+version="${VERSION:=$3}"
+#shellcheck disable=SC2153
+url="${URL:-$4}"
+bosh_secrets="${BOSH_SECRETS:-$5}"
+
+[[ -z "${type}" ]] && echo "Must provide artifact type: release|stemcell" && exit 100
+[[ -z "${name}" ]] && echo "Must provide artifact name" && exit 100
+[[ -z "${version}" ]] && echo "Must provide artifact version" && exit 100
+[[ -z "${url}" ]] && echo "Must provide download url" && exit 100
+[[ -z "${bosh_secrets}" ]] && echo "Must provide bosh secrets file path" && exit 100
+
+bosh_password=$(awk '$1~/bosh_admin_password/ {print $2}' "${bosh_secrets}")
```

Comment:
As we were discussing in person, I am not convince that we do this `awk` on the `bosh-secrets.yml` in this script itself.

I think that we should do it outside of the script, and only do the  `awk` in the pipeline. This way we decouple the method we use to retrieve the password from this script.